### PR TITLE
iOS-1681 | Sharing extension

### PR DESCRIPTION
**Pull request by bududomasidet**

### CODE REVIEW

File: Anytype/Sources/Design system/InlinePicker/AnytypeInlinePicker.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    @State private var initialValue: T?
+    private var allValues: [T]
+    private let selectionHandler: (T) -> Void
+    
+    public init(allValues: [T], selectionHandler: @escaping (T) -> Void) {
+        self.allValues = allValues
+        self.selectionHandler = selectionHandler
+    }
+    
+    @MainActor public var body: some View {
```
Comment (mgolovko) [inline Anytype/Sources/Design system/InlinePicker/AnytypeInlinePicker.swift:17]: body is mainActor by default

File: Anytype/Sources/PresentationLayer/ShareView/ShareModuleAssembly.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            return nil
+        }
+        
+        let viewModel = ShareViewModel(
+            contentViewModel: contentViewModel,
+            interactor: serviceLocator.sharedContentInteractor,
+            contentManager: serviceLocator.sharedContentManager
+        )
+    
+        viewModel.onClose = onClose
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ShareView/ShareModuleAssembly.swift:57]: Why do you setup part from init and part from properties?

File: Anytype/Sources/PresentationLayer/ShareView/ShareViewModelInteractor.swift [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                targetId: lastBlockInDocument,
+                info: blockInformation,
+                position: .bottom
+            )
+        }
+    }
+}
+
+
+extension URL {
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ShareView/ShareViewModelInteractor.swift:130]: maybe mark this extensions as private?

File: Anytype/Sources/PresentationLayer/ShareView/SharedContentManager.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    case image(URL)
+}
+
+final class SharedContentManager: SharedContentManagerProtocol {
+    private lazy var encoder = JSONEncoder()
+    private lazy var decoder = JSONDecoder()
+    private lazy var userDefaults = UserDefaults(suiteName: TargetsConstants.appGroup)
+    
+    func saveSharedContent(content: [SharedContent]) throws {
+        let sharedData = try JSONEncoder().encode(content)
```
Comment (mgolovko) [inline Anytype/Sources/PresentationLayer/ShareView/SharedContentManager.swift:21]: `try encoder.encode(content)`?

File: Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift [hunk 5]
```diff
@@ -3,8 +3,11 @@ import Combine
 import Services
 import AnytypeCore
 
-protocol SearchServiceProtocol: AnyObject {
+protocol SearchInteractorProtocol: AnyObject {
```
Comment (mgolovko) [inline Anytype/Sources/ServiceLayer/Object/Search/SearchService.swift:6]: why Interactor? this is service 🤔

File: Anytype/Sources/PresentationLayer/ShareView/ShareModuleAssembly.swift (hunk 2 above)
Comment (bududomasidet) [reply Anytype/Sources/PresentationLayer/ShareView/ShareModuleAssembly.swift:57]: I believe it's essential to encapsulate routing logic separately from assembly concerns. There are instances when we may need to invoke a method on the viewModel after a routing operation, and I don't believe assembly should be responsible for handling such tasks.

**Pull request merged**